src/networkwatcher.cpp

```diff
@@ -174,3 +175,31 @@ QString NetworkWatcher::getCurrentTransport() {
   return QString(metaEnum.valueToKey(type))
       .remove("TransportType_", Qt::CaseSensitive);
 }
+
+void NetworkWatcher::setImpl(NetworkWatcherImpl* networkWatcherImpl) {
+  logger.debug() << "setImpl";
+
+  m_impl = new InspectorNetworkWatcher(this);
```

Comment:
Ohnoo sorry i completly missed that in my first review.  😭 
Now as this class creates and owns the NetworkWatcherImpl, it should also cleanup. 

So if m_impl is already containing an object it probably should delete that. Otherwise the old one may still send signals to the Watcher. 


so something like `if(m_impl) m_impl->deleteLater()` or putting m_impl in a smart pointer i.e `std::unique_ptr`